modules/codegen/src/main/scala/com/twilio/guardrail/terms/SwaggerTerm.scala

```diff
@@ -71,17 +72,29 @@ case class SecurityRequirements(requirements: NonEmptyList[NonEmptyMap[String, S
                                 location: SecurityRequirements.Location)
 
 case class RouteMeta(path: String, method: HttpMethod, operation: Operation, securityRequirements: Option[SecurityRequirements]) {
+  object MediaType {
+    def unapply(value: MediaType): Option[(Option[Schema[_]], Option[Map[String, Encoding]], Option[Map[String, Object]])] = {
+      val schema: Option[Schema[_]] = Option(value.getSchema)
+      val encoding                  = Option(value.getEncoding()).map(_.asScala.toMap)
+      val extensions                = Option(value.getExtensions()).map(_.asScala.toMap)
+      Option((schema, encoding, extensions))
```

Comment:
Is it worth defining a case class to make this 3-tuple clearer?

Reply:
It wouldn't make it clearer, the only way it's even possible to be used is in the local scope, and it's just used in the `case MediaType(_, _, _) =>`. Changing it to `case MediaType(MediaTypeHolder(_, _, _)) =>` isn't clearer, imo.

That would be a useful technique to create safe extractors for OpenAPI types, but I don't believe `unapply` is the place to do that conversion -- it's too magical, and introduces a completely idiosyncratic pattern.

It may be useful for readability in the future, but right now it would be too big of a shift, both stylistically and mentally.

Reply:
> the only way it's even possible to be used is in the local scope

a quick follow-up: I found an edge case with companion objects defined for classes that don't have them and are defined in different packages. I tried defining this object in a shared location and importing it, but I got import conflicts. I think this is an edge case in scalac, but I don't care to investigate currently.